### PR TITLE
Added UT cases for start_indexing and indexing

### PR DESCRIPTION
**Pull request by Chenliang-Eric-Li**
Added ut cases for master_application.start_indexing task.iterate_dir and task.indexing in test_datamover.py.
Mocked nfs_cluster class, os.Entry class and some os functions in conftest.py #64

### CODE REVIEW

**Review by nsarras — CHANGES_REQUESTED** [review]
great job, just need to make a few changes

File: dss_datamover/tests/conftest.py [hunk 1]
```diff
@@ -52,7 +68,8 @@ def get_pytest_configs():
 
 @pytest.fixture(scope="session")
 def get_config_object():
-    test_config_filepath = os.path.dirname(__file__) + "/pytest_config.json"
+    # test_config_filepath = os.path.dirname(__file__) + "/pytest_config.json"
+    test_config_filepath = "/etc/dss/datamover/standard_config.json"
```
[the remark this answers is not in the document]
Comment (Chenliang-Eric-Li) [reply dss_datamover/tests/conftest.py]: Removed this line and put all the configures in pytest_config.json.

File: dss_datamover/tests/conftest.py [hunk 2]
```diff
@@ -103,19 +121,144 @@ class MockSocket():
     """
     Dummy Object for an actual socket, should simulate all basic functions of a socket object
     """
-    # TODO: finish building out MockSocket class
+    def __init__(self, family=0, type=0, proto=0, fileno=0):
```
[the remark this answers is not in the document]
Comment (Chenliang-Eric-Li) [reply dss_datamover/tests/conftest.py]: restored MockSocket class to its original state.

File: dss_datamover/tests/conftest.py [hunk 3]
```diff
@@ -179,3 +330,135 @@ def _method(master):
...
+
+@pytest.fixture
+def get_indexing_kwargs():
+    return {"data": "", "task_queue": Queue(), "logger": MockLogger(), "index_data_queue": Queue(),
+            "nfs_cluster": "", "nfs_share": "", "max_index_size": 2, "indexing_started_flag": Value('i', 0),
+            "index_data_count": Value('i', 0), "index_msg_count": Value('i', 0), "index_data_queue_size": 10000, "progress_of_indexing": {},
+            "prefix_index_data": {}, "standalone": False, "params": {"s3config": "dss_client", "dryrun": False, "resume_flag": False}}
+
+
+class MockEntry():
```
[the remark this answers is not in the document]
Comment (Chenliang-Eric-Li) [reply dss_datamover/tests/conftest.py]: it's called MockDirEntry now.

File: dss_datamover/tests/conftest.py [hunk 4]
```diff
@@ -179,3 +330,135 @@ def _method(master):
...
+        self.name = name
+
+    def stat(self):
+        return self.stat_
+
+    def is_dir(self):
+        return False
+
+
+def mock_os_scan_dir(dir):
```
[the remark this answers is not in the document]
Comment (Chenliang-Eric-Li) [reply dss_datamover/tests/conftest.py]: they are in a MockOSDirOperations class now.

**Review by nsarras — APPROVED** [review]

**Comment by nsarras** [comment]
Please pull in latest changes from master and conform to the new test structure

**Comment by nsarras** [comment]
@Chenliang-Eric-Li branch conflicts need to be resolved

**Comment by nsarras** [comment]
@Chenliang-Eric-Li is this a copy of another PR? we need to focus on one at a time and make sure there are no overlapping changes

**Pull request closed without merging**